src/transformStateWithClones.js

```diff
@@ -1,13 +1,40 @@
 'use strict';
-
-/**
+/*
  * @param {Object} state
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const updateState = { ...state };
+  const updateActions = [];
+
+  for (const key of actions) {
+    switch (key.type) {
+      case 'addProperties':
+        Object.assign(updateState, key.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const item of key.keysToRemove) {
+          delete updateState[item];
+        }
+        break;
+
+      case 'clear':
+        for (const a in updateState) {
+          delete updateState[a];
+        }
+        break;
+
+      default:
+        break;
```

Comment:
[CODE STYLE]: Use a `switch` statement for handling different action types, which is done correctly here. However, ensure that the `default` case includes error handling or a comment explaining why no action is necessary.

Reply:
> [CODE STYLE]: Use a `switch` statement for handling different action types, which is done correctly here. However, ensure that the `default` case includes error handling or a comment explaining why no action is necessary.

Good point